File: Spring/project1/demo/demo/src/main/plane/redisrepo/Aircraft.java
package main.plane.redisrepo;

import com.example.app.Droid;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
//import javax.persistence.Id;
//import javax.persistence.Entity;
import java.time.Instant;

//@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class Aircraft {

//    @Id
//    public long id;

    private String callsign, squwak, reg, flightno, route, type, category;
    private int altitude, heading, speed;

    @JsonProperty("vert_rate")
    private int vertRate;
    @JsonProperty("selected_altitude")
    private int selectedAltitude;
    private double lat, lon, barometer;
    @JsonProperty("polar_distance")
    private double polarDistance;
    @JsonProperty("polar_bearing")
    private double polarBearing;

    @JsonProperty("is_adsb")
    private boolean isADSB;
    @JsonProperty("is_on_ground")
    private boolean isOnGround;

    @JsonProperty("last_seen_time")
    private Instant lastSeenTime;
    @JsonProperty("pos_update_time")
    private Instant posUpdateTime;
    @JsonProperty("bds40_seen_time")
    private Instant bds40SeenTime;

    public String getLastSeemTime(){
        return lastSeenTime.toString();
    }

    public void setLastSeenTime(String lastSeenTime){
        if(null != lastSeenTime){
            this.lastSeenTime = Instant.parse(lastSeenTime);
        } else{
            this.lastSeenTime = Instant.ofEpochSecond(0);
        }
    }

    public String getPostUpdateTime(){
        return posUpdateTime.toString();
    }

    public void setPostUpdateTime(String posUpdateTime){
        if(null != posUpdateTime){
            this.posUpdateTime =Instant.parse(posUpdateTime);
        } else{
            this.posUpdateTime = Instant.ofEpochSecond(0);
        }
    }

    public Instant getBds40SeenTime() {
        return bds40SeenTime;
    }

    public void setBds40SeenTime(String bds40SeenTime) {
        if(null != bds40SeenTime){
            this.bds40SeenTime =Instant.parse(bds40SeenTime);
        } else{
            this.bds40SeenTime = Instant.ofEpochSecond(0);
        }
    }
}

File: Spring/project1/app/app/src/main/kotlin/plane/Aircraft.kt
package main.plane.mongodb

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.Instant

@Document
@JsonIgnoreProperties(ignoreUnknown = true)
data class Aircraft(
    @Id val id: String,
    val callsign: String? = "",
    val sqwak: String? = "",
    val reg: String? = "",
    val flightno: String? = "",
    val route: String? = "",
    val type: String? = "",
    val category: String? = "",
    val altitude: Int? = 0,
    val heading: Int? = 0,
    val speed: Int? = 0,
    @JsonProperty("vert_rate") val verRate: Int? = 0,
    @JsonProperty("selected_altitude")
    val selectedAltitude: Int? = 0,
    val lat: Double? = 0.0,
    val lon: Double? = 0.0,
    val barometer: Double? = 0.0,
    @JsonProperty("polar_distance")
    val polarDisatnce: Double? = 0.0,
    @JsonProperty("polar_bearing")
    val polarBearing: Double? = 0.0,
    @JsonProperty("is_adsb")
    val isADSB: Boolean? = false,
    @JsonProperty("is_on_groung")
    val isOnGround: Boolean? = false,
    @JsonProperty("last_seen_time")
    val lastSeenTime: Instant? = Instant.ofEpochSecond(0),
    @JsonProperty("pos_update_time")
    var posUpdateTime: Instant? = Instant.ofEpochSecond(0),
    @JsonProperty("bds40_seen_time")
    var bds40SeenTime: Instant? = Instant.ofEpochSecond(0)
)